### ISSUE TITLE
cv2.VideoCapture() is not working when given URL, but its working on python IDE.

### ISSUE THREAD

**Issue by MaujMishra** [pyscript/pyscript#1085]
### Checklist

- [X] I added a descriptive title
- [X] I searched for other issues and couldn't find a solution or duplication
- [X] I already searched in Google and didn't find any good information or help

### What happened?

I am building a GUI with pyscript, I installed opencv along with other libraries. I have uploaded a video on drive, I need to access that video in the GUI and run some processing on the frames. Here's my HTML - 

`<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="theme-color" content="#000000" />
    
    <meta
      name="description"
      content="Web site created using create-react-app"
    />
    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
 
    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
    <title>React App</title>
    <link rel="stylesheet" href="https://pyscript.net/latest/pyscript.css" />
    <script defer src="https://pyscript.net/latest/pyscript.js"></script>
  </head>
  <body>
        
    <py-config>
      packages = ["keras", "opencv-python", "numpy", "pandas"]
    </py-config>
    <noscript>You need to enable JavaScript to run this app.</noscript>
    <div id="root"></div>
  
    <py-script src="pred.py"></py-script>

  </body>
</html>
`

Here is code for python file - 

`import pandas as pd
import cv2

url = ("https://drive.google.com/uc?id=1BBJ0-mV89DNGYYETLB5ZDUG8LGQrXec_")

vidcap = cv2.VideoCapture(url)
count = 0

while(True):
    res, frame = vidcap.read()
    if res is False:
        break
    else:
        count += 1
print(count)`

This outputs - 

0

Which is not correct, But when I run the same code in the python IDE. The output is - 

911

Which is correct, What seems to be the issue?

### What browsers are you seeing the problem on? (if applicable)

Chrome

**Comment by FabioRosado** [comment]
Hello @MaujMishra thank you for raising this issue.

I played a bit with your example, but wasn't able to find any information to help you here.  I don't really know opencv, and there's no information about any failure to fetch the video from the URL.

After looking at python-opencv I noticed a test that would use a video and try to check if the video would be open with:

```python
url = ("https://drive.google.com/uc?id=1BBJ0-mV89DNGYYETLB5ZDUG8LGQrXec_")
vidcap = cv2.VideoCapture(url, cv2.CAP_VFW)
assert vidcap.isOpened()
```

The assertion failed. I then looked at the emscripten port in pyodide just to check if i could get any information, but didn't get much. Although it did seem there is a check to see if ffmeg is installed.  So I tried to use the [ffmeg port for wasm](https://github.com/ffmpegwasm/ffmpeg.wasm) but that didn't work.

The only thing I can suggest is trying to figure out why the file isn't being opened (which I suspect this is what you have been doing). Maybe try to open the file and pass it to `VideoCapture` if that works 🤔

**Comment by MaujMishra** [comment]
Thanks for the response. 
The file can be opened when running the code in python IDE. But there seems to be some problem with the pyscript, because the code is not running in pyscript

**Comment by FabioRosado** [comment]
Right, that's what I meant. Since using the non emscripten version of the library works, the issue could be in the compiled version fr emscripten, for reference I believe [this is the package](https://github.com/pyodide/pyodide/tree/main/packages/opencv-python) that is being installed by pyodide.

**Comment by ryanking13** [comment]
I don't know much about OpenCV internals, but I guess opening a video from URL would require a socket connection to download a video. Unfortunately, socket connection is not available in Pyodide due to browser limitations (at least for now).
